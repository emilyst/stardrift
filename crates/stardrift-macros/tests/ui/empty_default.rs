// This test should fail to compile because the default attribute is empty
use stardrift_macros::ConfigDefaults;

#[derive(ConfigDefaults)]
struct EmptyDefault {
    #[default()] // Empty!
    pub field1: i32,
}

fn main() {}
