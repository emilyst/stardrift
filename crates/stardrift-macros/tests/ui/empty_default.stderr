error: Field 'field1' has an empty #[default()] attribute. Please provide a default value.
 --> tests/ui/empty_default.rs:6:5
  |
6 |     #[default()]  // Empty!
  |     ^^^^^^^^^^^^
